hack/pin-dependency.sh

```diff
@@ -42,10 +42,11 @@ dep="${1:-}"
 sha="${2:-}"
 if [[ -z "${dep}" || -z "${sha}" ]]; then
   echo "Usage:"
-  echo "  hack/pin-dependency.sh \$MODULE \$SHA-OR-TAG"
+  echo "  hack/pin-dependency.sh \$MODULE[=\$REPLACEMENT] \$SHA-OR-TAG"
   echo ""
-  echo "Example:"
+  echo "Examples:"
   echo "  hack/pin-dependency.sh github.com/docker/docker 501cb131a7b7"
+  echo "  hack/pin-dependency.sh github.com/docker/docker=github.com/johndoe/docker my-experimental-branch"
```

Comment:
while this is handy for testing, we would not merge a replacement like this... can we make that clear in the help?